lib/json/json.go

```diff
@@ -100,8 +102,19 @@ func encode(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, k
 		}
 	}
 
+	path := make([]unsafe.Pointer, 0, 8)
+
 	var emit func(x starlark.Value) error
 	emit = func(x starlark.Value) error {
+		if ptr := pointer(x); ptr != nil {
+			if pathContains(path, ptr) {
+				return fmt.Errorf("Detected cycle in json structure")
```

Comment:
"cycle in..."

(no need for "Detected"; also lower case)

Reply:
Done